osu.Game/Graphics/Carousel/Carousel.cs

```diff
@@ -234,27 +234,27 @@ protected virtual void HandleFilterCompleted()
             Scroll.Panels.SingleOrDefault(p => ((ICarouselPanel)p).Item == item);
 
         /// <summary>
-        /// When a user is traversing the carousel via group selection keys, assert whether the item provided is a valid target.
+        /// When a user is traversing the carousel via set selection keys, assert whether the item provided is a valid target.
         /// </summary>
         /// <param name="item">The candidate item.</param>
-        /// <returns>Whether the provided item is a valid group target. If <c>false</c>, more panels will be checked in the user's requested direction until a valid target is found.</returns>
-        protected virtual bool CheckValidForGroupSelection(CarouselItem item) => true;
+        /// <returns>Whether the provided item is a valid set target. If <c>false</c>, more panels will be checked in the user's requested direction until a valid target is found.</returns>
+        protected virtual bool CheckValidForSetSelection(CarouselItem item) => true;
```

Comment:
I cannot stop myself from remarking that thus far the facade of an abstract "carousel of things" that was separate from beatmap carousel (which is likely the only carousel that will ever exist) was *sort of* holding, but now the "set" vernacular, which is highly specific to beatmap carousel and does not translate to a general "carousel of things", has crept its way into the base "carousel of things", and only because of wanting to replicate stable's weird keyboard behaviour too.

Maybe this could be salvaged if `traverseSetSelection()` along with `Activate{Previous,Next}Set` global actions was moved to `BeatmapCarousel`. Or maybe this is splitting hairs because the classes being split never made any semantic sense and was only arbitrary for some semblance of logic separation.

Reply:
Given your ongoing commentary that the split doesn't make sense and you don't care, I was aware of this fact but didn't give it much thought. One day it may make sense to combine them, but right now – at least for me – the separation of duties between the two classes is making the code more manageable.